### PR TITLE
Deadend radical algorithm RMG-RMS twin PR

### PR DESCRIPTION
**Pull request by hwpang**
This is a twin PR with https://github.com/ReactionMechanismGenerator/RMG-Py/pull/2388.

### CODE REVIEW

File: src/EdgeAnalysis.jl [hunk 1]
```diff
@@ -305,18 +309,28 @@ function processfluxes(sim::SystemSimulation,
             if @inbounds any(d.rxnarray[:,i].>length(corespeciesconcentrations))
                 continue
             end
+            net_consumption = max(frts[i+index]-rrts[i+index], 0.0)
```
Comment (mjohnson541) [inline src/EdgeAnalysis.jl:312]: maybe rename

Comment (mjohnson541) [reply src/EdgeAnalysis.jl:312]: net_forward_rate or something like that

File: src/EdgeAnalysis.jl [hunk 2]
```diff
@@ -490,6 +490,69 @@ end
...
+        else
+            reactantside = productinds[:,index]
+            productside = reactantinds[:,index]
+        end
+            
+        productrade = [sim.species[i].radicalelectrons for i in productside if i != 0]
+        reactantrade = [sim.species[i].radicalelectrons for i in reactantside if i != 0]
+
+        if length(productrade) == 1 && length(reactantrade) == 2
+            if productrade[1] == 0 && reactantrade[1] == 1 && reactantrade[2] == 1
```
Comment (mjohnson541) [inline src/EdgeAnalysis.jl:514]: Something here productrade and reactantrade  should never be 0 so why do we have checks for that?

File: src/EdgeAnalysis.jl [hunk 3]
```diff
@@ -305,18 +309,28 @@ function processfluxes(sim::SystemSimulation,
             if @inbounds any(d.rxnarray[:,i].>length(corespeciesconcentrations))
                 continue
             end
+            net_consumption = max(frts[i+index]-rrts[i+index], 0.0)
             for j = 1:4
                 if @inbounds d.rxnarray[j,i] != 0
                     @inbounds corespeciesconsumptionrates[d.rxnarray[j,i]] += frts[i+index]
                     @inbounds corespeciesproductionrates[d.rxnarray[j,i]] += rrts[i+index]
+                    if d.species[d.rxnarray[j,i]].radicalelectrons == 1
+                        coreradicalnetconsumptionrates[d.rxnarray[j,i]] += net_consumption
```
Comment (mjohnson541) [inline src/EdgeAnalysis.jl:318]: rname and do consumption vector for all species?

File: src/TestEdgeAnalysis.jl [hunk 4]
```diff
@@ -55,7 +55,7 @@ using Sundials
     reactedge = Reactor(coreedgedomain,coreedgey0,(0.0,1e6);p=coreedgep);
     (terminated,resurrected,invalidobjects,unimolecularthreshold,bimolecularthreshold,
     trimolecularthreshold,maxedgespeciesrateratios) = selectobjects(react,reactedge,coreedgedomain,[],coredomain,
-        [],corep,coreedgep,0.03,0.03,false,true,5,0.005,1.0,1.0,true,termination,1.0e8,Dict(),20)
+        [],corep,coreedgep,0.03,0.03,false,true,5,0.005,1.0,1.0,true,termination,1.0e8,Dict(),20,1e-2)
```
Comment (mjohnson541) [inline src/TestEdgeAnalysis.jl:58]: Maybe just Inf?

**Review by mjohnson541 — CHANGES_REQUESTED** [review]
Looks good, a few comments!

File: src/EdgeAnalysis.jl (hunk 3 above)
Comment (hwpang) [reply src/EdgeAnalysis.jl:318]: Done

File: src/EdgeAnalysis.jl [hunk 5]
```diff
@@ -820,6 +855,8 @@ function identifyobjects!(sim,corespcsinds,corerxninds,edgespcsinds,
                     @inbounds spcname = transitoryoutdict[ind]
                     @inbounds tol = transitorydict[spcname]
                     @info "at time $t sec, reaction $rstr at a normalized transitory sensitivity from $spcname of $sens exceeded the threshold of $tol for moving to model core"
+                elseif new_object_type == "deadendradical"
+                    @info "at time $t sec, reaction $rstr at a net radical consumption/termination ratio of $val exceeded the threshold of $val for moving to model core"
```
[the remark this answers is not in the document]
Comment (hwpang) [reply src/EdgeAnalysis.jl:859]: Done

**Review by mjohnson541 — APPROVED** [review]

**Review by mjohnson541 — APPROVED** [review]

**Pull request merged**